frameworks/Java/javalin/javalin.dockerfile

```diff
@@ -0,0 +1,7 @@
+FROM gradle:4.7.0-jdk8
+USER root
+WORKDIR /javalin
+COPY build.gradle build.gradle
+COPY ./ ./
+COPY src src
+CMD ["gradle", "--no-daemon", "--refresh-dependencies", "run", "-Pargs=8080,production"]
```

Comment:
Can you look into whether it's possible to separate this into a `gradle build` step and a `java -jar ...` step?  See some of the other Java frameworks that use Gradle as an example.

We'd make better use of Docker's caching this way.  As you have it now, it does not cache the result of compiling the code so it'll do that on every run even if there was no change to the implementation.  Also, I don't know exactly what `--refresh-dependencies` does, but that definitely sounds like something we don't want to happen in a `CMD` (every time we benchmark the framework) and perhaps not at all (if running this code now pulls version X of a dependency, and running it later will pull version X+1 of that same dependency, that introduces unwanted instability to our builds).

Reply:
I mostly took after the jawn implementation. Made the changes.